# Case study on testing of hypothesis

In [2]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import chi2_contingency
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
#Let us load the data
df=pd.read_csv('Sales_add.csv')

In [4]:
df.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [5]:
df.isna().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

In [6]:
df.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

In [32]:
df[['Region','Manager']].value_counts(ascending=True)

Region      Manager    
Region - B  Manager - B    1
Region - C  Manager - A    1
            Manager - C    1
Region - B  Manager - C    2
Region - A  Manager - B    3
            Manager - C    3
Region - C  Manager - B    3
Region - A  Manager - A    4
Region - B  Manager - A    4
dtype: int64

In [8]:
df.shape

(22, 5)

In [9]:
df.groupby(['Region'])['Sales_before_digital_add(in $)','Sales_After_digital_add(in $)'].mean()

<ipython-input-9-0cc1f487a21f>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df.groupby(['Region'])['Sales_before_digital_add(in $)','Sales_After_digital_add(in $)'].mean()


Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
Region                                                                   
Region - A                   148204.900000                  238853.100000
Region - B                   150523.571429                  228727.857143
Region - C                   149513.000000                  219019.200000

# The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

Null hypothesis H0: sales after digital marketing <= sales before digital marketing

Alternative hypothesis H1: sales after digital marketing > sales before digital marketing

In [10]:
before_digital_marketing=df['Sales_before_digital_add(in $)']

In [11]:
before_digital_marketing

0     132921
1     149559
2     146278
3     152167
4     159525
5     137163
6     130625
7     131140
8     171259
9     141956
10    159339
11    178939
12    145062
13    151514
14    147463
15    177195
16    140862
17    167996
18    132135
19    152493
20    147425
21    130263
Name: Sales_before_digital_add(in $), dtype: int64

In [12]:
after_digital_marketing=df['Sales_After_digital_add(in $)']

In [13]:
after_digital_marketing

0     270390
1     223334
2     244243
3     231808
4     258402
5     256948
6     222106
7     230637
8     226261
9     193735
10    203175
11    276279
12    205180
13    253131
14    229336
15    187305
16    234908
17    191517
18    227040
19    212579
20    263388
21    243020
Name: Sales_After_digital_add(in $), dtype: int64

In [14]:
t_stat,p_value=stats.ttest_ind(after_digital_marketing,before_digital_marketing,alternative='greater')

In [15]:
print('The value for t-statistics is: ',t_stat, 'The value for p-value is: ',p_value)

The value for t-statistics is:  12.995084451110877 The value for p-value is:  1.3071840034523225e-16


Let us assume that alpha value is 0.05, from the above result we can see that we have got lesser p-value. Which is less than alpha value. So, we'll reject the null hypothesis. 

Conclusion:-Hence, we conclude that there is increase in sales after digital marketing

# The company needs to check whether there is any dependency between the features “Region” and “Manager”.

Null hypothesis: H0: Region and Manager are independent

Alternative hypothesis: H1: Region and Manager are dependent

In [16]:
contingency=pd.crosstab(df['Region'],df['Manager'],margins=True)

In [17]:
contingency

Manager     Manager - A  Manager - B  Manager - C  All
Region                                                
Region - A            4            3            3   10
Region - B            4            1            2    7
Region - C            1            3            1    5
All                   9            7            6   22

In [18]:
stats.chi2_contingency(contingency)

(3.050566893424036,
 0.962256341757093,
 9,
 array([[ 4.09090909,  3.18181818,  2.72727273, 10.        ],
        [ 2.86363636,  2.22727273,  1.90909091,  7.        ],
        [ 2.04545455,  1.59090909,  1.36363636,  5.        ],
        [ 9.        ,  7.        ,  6.        , 22.        ]]))

Degree of freedom is 9, p-value is 0.962 and x^2 stat value is 3.05. P-value is higher than alpha value= 0.05

From the above result, we can accept null hypothesis. 

Conclusion:-Manager and region are independent. 